spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorHandlerThrowingException.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.errorhandling.implementations
+
+import org.apache.spark.sql.functions.{coalesce, udf, when}
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.types.DataType
+import za.co.absa.spark.commons.errorhandling.partials.TransformIntoErrorMessage.FieldNames._
+import za.co.absa.spark.commons.errorhandling.{ErrorHandling, ErrorMessage}
+import za.co.absa.spark.commons.errorhandling.partials.TransformIntoErrorMessage
+import za.co.absa.spark.commons.sql.functions.null_col
+
+object ErrorHandlerThrowingException extends ErrorHandling with TransformIntoErrorMessage {
+  case class ErrorHandlerException(error: ErrorMessage) extends Exception(error.errMsg)
+
+  /**
+   * @see [[ErrorHandling.doApplyErrorColumnsToDataFrame]]
+   */
+  override protected def doApplyErrorColumnsToDataFrame(dataFrame: DataFrame, errCols: Column*): DataFrame = {
+    import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
+    val errorColumn= coalesce(errCols: _*)
+    val errColName = dataFrame.schema.getClosestUniqueName("_1")
```

Comment:
so the column name will be `_1_1` - a bit odd although I know that it's temporary and will be almost immediately dropped. But if there is an error and we need to debug it, it can be better to have a bit more descriptive name still. But I'll leave it up to you

Reply:
Odd is good. It avoids name clashes 😉 
What name would you suggest?

Reply:
How about 1temporaryErrCol`?

Reply:
OK, went with more more SQL standard and prefixing an underscore, which is a typical sign of "more private" field: `_temporary_err_col`.